bundles/org.openhab.binding.dirigera/src/main/java/org/openhab/binding/dirigera/internal/handler/plug/PowerPlugHandler.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.dirigera.internal.handler.plug;
+
+import static org.openhab.binding.dirigera.internal.Constants.*;
+
+import java.util.Iterator;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.json.JSONObject;
+import org.openhab.binding.dirigera.internal.interfaces.Model;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.types.Command;
+
+/**
+ * The {@link PowerPlugHandler} basic DeviceHandler for all devices
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class PowerPlugHandler extends SimplePlugHandler {
+    public PowerPlugHandler(Thing thing, Map<String, String> mapping) {
+        super(thing, mapping);
+        super.setChildHandler(this);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        // update of values is handled in super class
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        super.handleCommand(channelUID, command);
+        String channel = channelUID.getIdWithoutGroup();
+        String targetProperty = channel2PropertyMap.get(channel);
+        if (targetProperty != null) {
+            switch (channel) {
+                case CHANNEL_CHILD_LOCK:
+                case CHANNEL_DISABLE_STATUS_LIGHT:
+                    if (command instanceof OnOffType onOff) {
+                        JSONObject attributes = new JSONObject();
+                        attributes.put(targetProperty, onOff.equals(OnOffType.ON));
```

Comment:
A nit: best to have the constant on the left, it would prevent null pointer excpetions. Due to the instanceof that will never occur here. But better to be consistent  / defensive coding.
```suggestion
                        attributes.put(targetProperty, OnOffType.ON.equals(onOff));
```

Reply:
Changed on several locations